Phase01Project1/src/SearchFilters/SearchFilter.java

```diff
@@ -0,0 +1,21 @@
+package SearchFilters;
+
+import java.util.ArrayList;
+import java.util.Map;
+import Documents.Document;
+
+public abstract class SearchFilter {
+
+    protected ArrayList<String> filterWords ;
+    protected Map<String, ArrayList<Document>> wordDocuments;
+    protected ArrayList<Document> resultDocuments;
+    public SearchFilter(ArrayList<String> filterWords, Map<String, ArrayList<Document>> wordDocuments, ArrayList<Document> resultDocuments){
+        this.filterWords = filterWords;
+        this.wordDocuments = wordDocuments;
+        this.resultDocuments = resultDocuments;
+    }
+
+    abstract public ArrayList<Document> handleFilterWords();
+    abstract public ArrayList<Document> applyToResult();
+
+}
```

Comment:
it's a good act for Open/Closed but the naming of abstract methods is not specific and readable.